tests/functional/src/main/java/org/mule/functional/classloading/isolation/classloader/MuleClassLoaderIsolatedFactory.java

```diff
@@ -241,4 +221,24 @@ private Boolean isVerboseClassLoading()
         return valueOf(getProperty(MULE_LOG_VERBOSE_CLASSLOADING));
     }
 
+    /**
+     * Validates that only one module should be discovered. A plugin cannot have inside another plugin that holds a {@code mule-module.properties} for the time being.
+     *
+     * @param pluginName the plugin name
+     * @param discoveredModules {@link MuleModule} discovered
+     * @return the first Module from the list due to there should be only one module.
+     */
+    private MuleModule validatePluginModule(String pluginName, List<MuleModule> discoveredModules)
+    {
+        if (discoveredModules.size() == 0)
+        {
+            throw new IllegalStateException(pluginName + " doesn't have in its classpath a mule-module.properties to define what packages and resources should expose");
```

Comment:
I know you just moved these lines, but try to use a constant to reference mule-module.properties file